app/src/main/java/com/duckduckgo/app/systemsearch/SystemSearchActivity.kt

```diff
@@ -391,15 +403,34 @@ class SystemSearchActivity : DuckDuckGoActivity() {
         omnibarTextInput.setSelection(query.length)
     }
 
-    private fun confirmDeleteSavedSite(savedSite: SavedSite) {
+    private fun confirmDeleteSavedSite(
+        savedSite: SavedSite,
+        oldPosition: Int,
+    ) {
         val message = getString(R.string.bookmarkDeleteConfirmationMessage, savedSite.title).html(this)
         Snackbar.make(
             binding.root,
             message,
             Snackbar.LENGTH_LONG,
         ).setAction(R.string.fireproofWebsiteSnackbarAction) {
-            viewModel.insertQuickAccessItem(savedSite)
-        }.show()
+            viewModel.undoDelete(savedSite, oldPosition)
+        }
+            .addCallback(
+                object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
+                    override fun onDismissed(
+                        transientBottomBar: Snackbar?,
+                        event: Int,
+                    ) {
+                        // when snackbar is not dismissed because of an action we want to
```

Comment:
ditto. Just keep this agnostic of what should happen, and move logic to viewmodel.